runtime/src/balances.rs

```diff
@@ -1322,14 +1418,15 @@ impl<T: Trait<I>, I: Instance + Clone + Eq> SignedExtension for TakeFees<T, I> {
             Ok(key) => key,
             Err(_) => return InvalidTransaction::BadProof.into(),
         };
-
-        if <T::Identity>::signing_key_charge_did(&encoded_transactor) {
+        let imbalance;
+        if let Some(did) = <Module<T, I>>::charge_fee_to_identity(&encoded_transactor) {
             sr_primitives::print("Charging fee to identity");
-            if !<T::Identity>::charge_poly(&encoded_transactor, fee) {
-                return InvalidTransaction::Payment.into();
-            }
+            imbalance = match <Module<T, I>>::withdraw_identity_balance(&did, fee) {
+                Ok(imbalance) => imbalance,
+                Err(_) => return InvalidTransaction::Payment.into(),
+            };
         } else {
-            let imbalance = match <Module<T, I>>::withdraw(
+            imbalance = match <Module<T, I>>::withdraw(
```

Comment:
Ditto: If you think that previous one is more readable, pls apply here the same.
Otherwise, just mark this as solved.